### DISCUSSION TITLE
Should we add CUPiD to the CTSM checkout?

### DISCUSSION THREAD

**Discussion by ekluzek** [ESCOMP/CTSM#2864] [Ideas]
CESM now has included CUPiD in it's checkout and I wonder if it should be included in the CTSM checkout. Since, it's included in the CESM checkout I assume there are ways to invoke it from a case. And I assume you can also just have CTSM diagnostics exercised. I also know that it's possible to run ILAMB with CUPiD and that capability might be really good for CTSM scientists.

@TeaganKing you are likely the expert here -- do you think this is something we should do now? Or is it premature?

**Comment by wwieder** [comment]
I'd suspect this is premature, unless the ILAMB interface is fully functional in CUPiD?  As you note @ekluzek, @TeaganKing is the one to ask on the status of this.

It also raises questions about how the LMWG is going to support / build other diagnostics within CUPiD? I've started the following branches off:
- ADF that's at least functional for making plots.  
- CUPiD that should (in principle) generate single variable time series and make plots using uxarray.

I'm optimistic that with a bit of more focused effort we can get something started that's more robust on this front, but am less sure on when (or by whom) this is most likely to occur?

**Reply by TeaganKing** [reply]
And yes, ILAMB is a bit less than fully functional at the moment in CUPiD.

**Reply by ekluzek** [reply]
OK, it seems clear that it's premature now, but once there's some work on ILAMB or the hackathon idea below it would be good to bring in.

This was a helpful discussion I'm glad we asked here..

**Comment by dlawrenncar** [comment]
I wonder if a mini Hackathon (half a day or something) could get the land
package over the hump from a skeleton to something real.  It would seem to
be easier to add variables / diagnostics on top of something with more in
place (though I admit I don't really know the current status).

**Reply by wwieder** [reply]
That's a good idea.  I kind of think the hackathon may be most productive if we have more bones on the skeleton and then use the larger group to work on flushing out more of the 'meat'?

**Reply by TeaganKing** [reply]
One thing that may be relevant here is that Mike and I are currently working on two PRs (CUPiD [#142](https://github.com/NCAR/CUPiD/pull/142) and [#141](https://github.com/NCAR/CUPiD/pull/141)) that will implement both 1) a link to the full ADF package and 2) a few key ADF plots will be reproduced directly in line in the jupyter book output. I think it probably would be most efficient to have a mini hackathon focused on ILAMB after this work is complete (we're aiming to wrap it up on Friday).

**Reply by ekluzek** [reply]
That sounds like a good timeline to think of. So if @TeaganKing can let us know in a few weeks when she's ready to start on ILAMB we could get interested people to figure out when to do the hackathon. Since, the hackathon is a short term thing I'd like to be part of it, especially to figure out how CUPiD will integrate into CESM/CTSM.

@olyson and @slevis-lmwg it's seems like both of you should be involved?

**Reply by TeaganKing** [reply]
That seems reasonable. I can keep you all in the loop on ADF progress.

**Reply by TeaganKing** [reply]
We will be having an ADF capability demonstration on Wednesday at 1-1:30pm in the Damon room (and remotely). I'll send you all the calendar link. I think it would probably be helpful for anyone involved in bringing in ILAMB to have a better sense of how another external diagnostic package is being brought in to CUPiD. FYI we will still have a few additional updates to make at the time of the demonstration. I think it would also be useful to get your feedback on Wednesday about what would/wouldn't work similarly with ILAMB.

**Comment by ekluzek** [comment]
I asked @TeaganKing an update on this and she sees just a few things that should happen before we do this. Here's the list of suggestions she had as requirements before this comes in:

> Hey Erik,
> 
> We should bring CUPiD into CTSM as a submodule once the following PRs are included in CUPiD:
> [Include ILAMB in key_metrics and CESM workflow](https://github.com/NCAR/CUPiD/pull/210/files) (needs a bit more work but getting close)
> [Enable LDF in CUPiD](https://github.com/NCAR/CUPiD/pull/226) (under final review)
> [Allow users to specify what CUPiD install to use in CESM Workflow](https://github.com/NCAR/CUPiD/issues/257) (is on Mike's high priority list)
> [Create a lnd-specific example](https://github.com/NCAR/CUPiD/issues/262) (This will be super quick & easy to do but hasn't been started)
> I'd guess this may take a month-ish before everything is ready...
> 
> Teagan


As such I'm going to close this issue and turn this comment into an issue that we'll start planning on.

**Comment by slevis-lmwg** [comment]
This is now active issue #3396
and active PR #3689 

Ok to close the discussion?
